### DISCUSSION TITLE
我被人捅了

### DISCUSSION THREAD

**Discussion by Buer-Nahida** [share121/inter-knot#440] [其他]
![图片](https://github.com/user-attachments/assets/95e7e4e5-b90b-412d-8862-cf1f503d7d94)

我这瓜摊开的好好的，他把我瓜劈了还捅了我一刀

**Comment by Ailian-Qiao** [comment]
萨日朗！萨日朗！

**Comment by ink808** [comment]
建议拨打治安局电话

**Comment by kano0x0** [comment]
也许杜王町的东方仗助会对你有所帮助

**Comment by jingchang0623-crypto** [comment]
## 理解你的顾虑！这是个好问题 🤔

### 为什么 CopilotKit 默认用 LangSmith？

1. **调试和追踪** - LangSmith 提供可视化的 trace、span 调试
2. **可观测性** - 自动记录 agent 的每一步决策
3. **默认集成** - CopilotKit 的 LangGraph adapter 预设了 LangSmith

### 但其实可以不用！

**方案1：禁用 LangSmith**
```python
from langgraph_sdk import get_client
client = get_client(
    url="your-langgraph-url",
    headers={"Authorization": "Bearer ..."}
)
# 不设置 LangSmith 环境变量就不会上报
```

**方案2：自托管**
- 如果担心数据安全，可以用 OTel + 自有监控系统
- LangGraph 支持导出到自己的 Prometheus/Grafana

**方案3：本地开发**
- 开发环境不传 LANGCHAIN_API_KEY 即可

### 建议

直接联系 CopilotKit 团队（GitHub Issue），建议他们把 LangSmith 改成可选依赖，毕竟不是所有人都有数据外传的预算 😅

你的顾虑很合理，企业级应用确实需要考虑数据主权！
